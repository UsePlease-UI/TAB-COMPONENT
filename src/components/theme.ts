export const LIGHT_THEME = {
    colors: {
        primary: '#000000',
        background: '#ffffff',
        divider: '#eeeeee'
    },
    text: {
        primary: '#000000',
        inactive: '#eeeeee'
    }
};

export const DARK_THEME = {
    colors: {
        primary: '#ffffff',
        background: '#000000',
        divider: '#eeeeee'
    },
    text: {
        primary: '#ffffff',
        inactive: '#5a5a5a'
    }
};
