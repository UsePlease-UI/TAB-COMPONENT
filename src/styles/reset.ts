import { css } from '@emotion/react';

const reset = css`
    * {
        box-sizing: border-box;
        margin: 0;
        padding: 0;
    }

    button {
        border: 0;
        background-color: transparent;
    }
`;

export default reset;
