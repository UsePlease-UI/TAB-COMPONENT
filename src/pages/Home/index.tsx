import { useState } from 'react';

import { Tab, TabList, TabItem, TabPanel } from 'components';
import TabContext from 'components/TabContext/provider';

const tabListStyles = {
    list: {
        height: '100px',
        backgroundColor: ''
    },
    item: {
        height: '100px'
    },
    divider: {
        dividerColor: 'transparent',
        dividerHeight: '5px'
    },
    label: {
        labelActiveColor: '',
        labelInactiveColor: '#eeeeee',
        labelLineHeight: '40px',
        labelFontSize: '40px',
        labelFontWeight: 400 as const,
        labelFontTracking: '0.5rem',
        labelFont: ''
    }
};

function getStyles(...args: Record<string, string | number>[]) {
    const styles = {};
    args.forEach((obj) => {
        Object.assign(
            styles,
            Object.keys(obj).reduce((prev, key) => ({ ...prev, [key]: obj[key] }), {})
        );
    });

    return styles;
}

export default function Home() {
    const [value, setValue] = useState(1);

    const handleClick = (e: React.SyntheticEvent, newValue: number) => {
        setValue(newValue);
    };

    return (
        <TabContext value={value} setValue={setValue}>
            <Tab>
                <TabList aria-label="navigation" styles={getStyles(tabListStyles.list, tabListStyles.divider)}>
                    <TabItem styles={getStyles(tabListStyles.item, tabListStyles.label)} label="1" value={value} index={1} onClick={handleClick} />
                    <TabItem styles={getStyles(tabListStyles.item, tabListStyles.label)} label="2" value={value} index={2} onClick={handleClick} />
                    <TabItem styles={getStyles(tabListStyles.item, tabListStyles.label)} label="3" value={value} index={3} onClick={handleClick} />
                </TabList>
                <TabPanel value={value} index={1}>
                    <p>Tab Panel #1</p>
                </TabPanel>
                <TabPanel value={value} index={2}>
                    <p>Tab Panel #2</p>
                </TabPanel>
                <TabPanel value={value} index={3}>
                    <p>Tab Panel #3</p>
                </TabPanel>
            </Tab>
        </TabContext>
    );
}
