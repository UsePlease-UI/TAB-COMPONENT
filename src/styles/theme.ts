import { LIGHT_THEME, DARK_THEME } from 'components/theme';

declare module '@emotion/react' {
    export interface Theme {
        colors: {
            primary: string;
            background: string;
            divider: string;
        };
        text: {
            primary: string;
            inactive: string;
        };
    }
}

export { DARK_THEME, LIGHT_THEME };
