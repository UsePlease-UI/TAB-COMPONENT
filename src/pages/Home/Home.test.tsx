import { ThemeProvider } from '@emotion/react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Home from 'pages/Home';

import { LIGHT_THEME } from 'styles/theme';

describe('<HOME />', () => {
    it('renders Home Component', () => {
        const tree = renderer
            .create(
                <ThemeProvider theme={LIGHT_THEME}>
                    <Home />
                </ThemeProvider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('if it is accessible when TabPanel#value === TabContext#value', () => {
        const { getByText } = render(
            <ThemeProvider theme={LIGHT_THEME}>
                <Home />
            </ThemeProvider>
        );

        expect(getByText('Tab Panel #1')).toBeInTheDocument();
    });

    test('if it is [hidden] when TabPanel#value !== TabContext#value', () => {
        const { getByText } = render(
            <ThemeProvider theme={LIGHT_THEME}>
                <Home />
            </ThemeProvider>
        );

        expect(getByText('Tab Panel #2')).not.toBeVisible();
    });
});
