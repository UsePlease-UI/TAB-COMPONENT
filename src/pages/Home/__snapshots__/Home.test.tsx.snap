// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HOME /> renders Home Component 1`] = `
<div
  className="css-1wirx3w"
>
  <div
    aria-label="navigation"
    className="css-1k39m3u"
    role="tablist"
  >
    <div
      className="css-9lc4tj"
    >
      <button
        aria-controls="tabpanel-1"
        aria-selected={true}
        className="css-1naocwp"
        id="tab-1"
        onClick={[Function]}
        role="tab"
        type="button"
      >
        <p
          className="css-1ph9szf"
        >
          1
        </p>
        <div
          className="css-3wo3jb"
          id="tab-indicator-1"
        />
      </button>
      <button
        aria-controls="tabpanel-2"
        aria-selected={false}
        className="css-1naocwp"
        id="tab-2"
        onClick={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <p
          className="css-1ybwqpv"
        >
          2
        </p>
      </button>
      <button
        aria-controls="tabpanel-3"
        aria-selected={false}
        className="css-1naocwp"
        id="tab-3"
        onClick={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <p
          className="css-1ybwqpv"
        >
          3
        </p>
      </button>
    </div>
  </div>
  <div
    className="css-bc2c4v"
    hidden={false}
    role="tabpanel"
  >
    <p>
      Tab Panel #1
    </p>
  </div>
  <div
    className="css-bc2c4v"
    hidden={true}
    role="tabpanel"
  >
    <p>
      Tab Panel #2
    </p>
  </div>
  <div
    className="css-bc2c4v"
    hidden={true}
    role="tabpanel"
  >
    <p>
      Tab Panel #3
    </p>
  </div>
</div>
`;
